internal/rbd/rbd_util.go

```diff
@@ -1363,9 +1364,36 @@ func lookupRBDImageMetadataStash(path string) (rbdImageMetadataStash, error) {
 	return imgMeta, nil
 }
 
+// updateRBDImageMetadataStash reads and updates stashFile with the required
+// fields at the passed in path, in JSON format.
+func updateRBDImageMetadataStash(metaDataPath, device string) error {
+	if device == "" {
+		return errors.New("device is empty")
+	}
+	imgMeta, err := lookupRBDImageMetadataStash(metaDataPath)
+	if err != nil {
+		return fmt.Errorf("failed to find image metadata: %w", err)
+	}
+	imgMeta.DevicePath = device
+
+	encodedBytes, err := json.Marshal(imgMeta)
+	if err != nil {
+		return fmt.Errorf("failed to marshal JSON image metadata for pool:(%s), image:(%s), radosnamespace:(%s) : %w",
+			imgMeta.Pool, imgMeta.ImageName, imgMeta.RadosNamespace, err)
```

Comment:
Define a string Interface on `imgMeta` like https://github.com/ceph/ceph-csi/blob/devel/internal/rbd/rbd_util.go#L202-L207 makes life simpler for logging